src/crawlers/puppeteer_crawler.js

```diff
@@ -100,20 +47,20 @@ import { openSessionPool } from '../session_pool/session_pool';
  *   If the function throws an exception, the crawler will try to re-crawl the
  *   request later, up to `option.maxRequestRetries` times.
  *   If all the retries fail, the crawler calls the function
- *   provided to the `options.handleFailedRequestFunction` parameter.
+ *   provided to the `handleFailedRequestFunction` parameter.
  *   To make this work, you should **always**
  *   let your function throw exceptions rather than catch them.
  *   The exceptions are logged to the request using the
  *   [`request.pushErrorMessage`](request#Request+pushErrorMessage) function.
- * @param {RequestList} options.requestList
+ * @property {RequestList} requestList
```

Comment:
while you need at least either `requestList` or `requestQueue`, right now both are required (TS wise), and won't compile

Reply:
done. Both `requestList` and `requestQueue` are now options. Docs make it clear, one of them must be provided.